java/src/main/java/cucumber/runtime/java/ObjectFactoryLoader.java

```diff
@@ -53,6 +55,46 @@ public static ObjectFactory loadObjectFactory(ClassFinder classFinder, String ob
         return objectFactory;
     }
 
+    private static io.cucumber.core.backend.ObjectFactory loadSingleObjectFactory(final Reflections reflections, final ServiceLoader<io.cucumber.core.backend.ObjectFactory> loader) {
```

Comment:
Rather then trying both the service loader and class path scanning this method should first try the service loader and then try class path scanning. Doing both and then checking the result makes more complicated then it needs to be.

Reply:
But than we do not detect if more than one ObjectFactory is available - currently it detect if relefection and service loader found different implementations.

Reply:
But can change it, but than I think in the other way? First Reflection than ServiceLoader?

Reply:
First ServiceLoader then Reflection. We're deprecating the reflection after all.